bundles/org.openhab.binding.fineoffsetweatherstation/src/main/java/org/openhab/binding/fineoffsetweatherstation/internal/discovery/FineOffsetGatewayDiscoveryService.java

```diff
@@ -168,15 +179,39 @@ private void discovered(String ip, int port, byte[] macAddr, String name) {
         properties.put(Thing.PROPERTY_MAC_ADDRESS, Utils.toHexString(macAddr, macAddr.length, ":"));
         properties.put(FineOffsetGatewayConfiguration.IP, ip);
         properties.put(FineOffsetGatewayConfiguration.PORT, port);
+        FineOffsetGatewayConfiguration config = new Configuration(properties).as(FineOffsetGatewayConfiguration.class);
+        Protocol protocol = determineProtocol(config);
+        if (protocol != null) {
+            properties.put(FineOffsetGatewayConfiguration.PROTOCOL, protocol.name());
+        }
 
         ThingUID uid = new ThingUID(THING_TYPE_GATEWAY, id);
         DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                .withRepresentationProperty(Thing.PROPERTY_MAC_ADDRESS)
```

Comment:
You might also need to add this:
```
<representation-property>macAddress</representation-property>
```
before this line:
https://github.com/openhab/openhab-addons/blob/c729b82d59473f1fac871a5b18bda35f697fcc7a/bundles/org.openhab.binding.fineoffsetweatherstation/src/main/resources/OH-INF/thing/gateway.xml#L11

Reply:
If it has any effect, the same might be needed in `sensor.xml` (property "sensor").